From 717a3c6fbe2f40cf1ff6c20f1f5e38314ccb9fec Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Mon, 31 Jan 2022 17:48:05 -0600
Subject: [PATCH 1/1] ARM:dts:aspeed: Add NC-SI configuration to all platform
 dts

1. NC_SI configuration added for all platforms
2. ftgmac100 code modified to read the alias from dts and assign the
intrface# accordingly

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts     | 12 ++++++++++--
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts   | 10 ++++++++++
 arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts     | 10 ++++++++++
 arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts | 10 ++++++++++
 drivers/net/ethernet/faraday/ftgmac100.c      |  8 +++++++-
 5 files changed, 47 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index 4b0bc1840772..0453d946e6dd 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -14,6 +14,8 @@

 	aliases {
 		serial4 = &uart5;
+		ethernet0 = &mac3;
+		ethernet1 = &mac2;
 	};

 	chosen {
@@ -78,6 +80,14 @@
 	};
 };

+&mac2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii3_default>;
+	clock-names = "MACCLK",	"RCLK";
+	use-ncsi;
+};
+
 &mac3 {
 	status = "okay";
 	phy-mode = "rgmii";
@@ -87,8 +97,6 @@
 	pinctrl-0 = <&pinctrl_rgmii4_default>;
 };

-
-
 &fmc {
 	status = "okay";
 	fmc-spi-user-mode;
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index aa67def7ece3..a0e01645593b 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -13,6 +13,8 @@

 	aliases {
 		serial4 = &uart5;
+		ethernet0 = &mac3;
+		ethernet1 = &mac2;
 	};

 	chosen {
@@ -77,6 +79,14 @@
 	};
 };

+&mac2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii3_default>;
+	clock-names = "MACCLK", "RCLK";
+	use-ncsi;
+};
+
 &mac3 {
 	status = "okay";
 	phy-mode = "rgmii";
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
index 975e0b254646..3be7877e1058 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
@@ -13,6 +13,8 @@

 	aliases {
 		serial4 = &uart5;
+		ethernet0 = &mac3;
+		ethernet1 = &mac2;
 	};

 	chosen {
@@ -78,6 +80,14 @@
 	};
 };

+&mac2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii3_default>;
+	clock-names = "MACCLK",	"RCLK";
+	use-ncsi;
+};
+
 &mac3 {
 	status = "okay";
 	phy-mode = "rgmii";
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts b/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
index b8e0caecf419..5610df002bb0 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
@@ -13,6 +13,8 @@

 	aliases {
 		serial4 = &uart5;
+		ethernet0 = &mac3;
+		ethernet1 = &mac2;
 	};

 	chosen {
@@ -76,6 +78,14 @@
 	};
 };

+&mac2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii3_default>;
+	clock-names = "MACCLK",	"RCLK";
+	use-ncsi;
+};
+
 &mac3 {
 	status = "okay";
 	phy-mode = "rgmii";
diff --git a/drivers/net/ethernet/faraday/ftgmac100.c b/drivers/net/ethernet/faraday/ftgmac100.c
index 1c7912a94e36..438a4a186983 100644
--- a/drivers/net/ethernet/faraday/ftgmac100.c
+++ b/drivers/net/ethernet/faraday/ftgmac100.c
@@ -1795,6 +1795,13 @@ static int ftgmac100_probe(struct platform_device *pdev)
 	priv->dev = &pdev->dev;
 	INIT_WORK(&priv->reset_task, ftgmac100_reset_task);

+	// read alias "ethernetX" from dts and assign to interface
+	pdev->id = of_alias_get_id(pdev->dev.of_node, "ethernet");
+	if (pdev->id < 0)
+		pdev->id = 0;
+
+	sprintf(netdev->name, "eth%d", pdev->id);
+
 	/* map io memory */
 	priv->res = request_mem_region(res->start, resource_size(res),
 				       dev_name(&pdev->dev));
@@ -1909,7 +1916,6 @@ static int ftgmac100_probe(struct platform_device *pdev)
 		dev_err(&pdev->dev, "Failed to register netdev\n");
 		goto err_register_netdev;
 	}
-
 	netdev_info(netdev, "irq %d, mapped at %p\n", netdev->irq, priv->base);

 	return 0;
--
2.17.1

